File: docs/competitors/Python/gcd/gcd.py
#GCD — читает два целых числа, печатает наибольший общий делитель (НОД, GCD) этих чисел
#использует алгоритм Евклида для поиска НОД

def gcd(a, b):
    a, b = abs(a), abs(b)
    if a < b:
        a, b = b, a
    while a % b != 0:
        a = a % b
        a, b = b, a
    return b

num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))
result = gcd(num1, num2)

print(f"GCD({num1}, {num2}) = {result}")

File: docs/competitors/C++/gcd/gcd.cpp
//GCD — читает два целых числа, печатает наибольший общий делитель (НОД, GCD) этих чисел
//использует алгоритм Евклида для поиска НОД

#include <iostream>
using namespace std;

int gcd(int a, int b) 
{
    if (a < b) 
    {
        swap(a, b);
    }
    while (a % b != 0) 
    {
        a = a % b;
        swap(a, b);
    }
    return b;
}

int main() 
{
    int num1, num2;

    cout << "Enter first number: ";
    cin >> num1;

    cout << "Enter second number: ";
    cin >> num2;

    int result = gcd(abs(num1), abs(num2));

    cout << "GCD(" << num1 << "," << num2 << ") = " << result << endl;

    return 0;
}